packages/eas-cli/src/build/ios/build.ts

```diff
@@ -66,6 +66,7 @@ export async function prepareIosBuildAsync(
   return await prepareBuildRequestForPlatformAsync({
     ctx: buildCtx,
     ensureCredentialsAsync: async (ctx: BuildContext<Platform.IOS>) => {
+      await setupPushKeyAsync(ctx, targets);
       return ensureIosCredentialsAsync(ctx, targets);
```

Comment:
If you decide not to go with my suggestion above, then change the order of those calls here, the push key should be created at the end.

Reply:
i went with your suggestion!